[FEAT] Automated audit logging

shatfield4 opened the pull request:
### Pull Request Type

<!-- For change type, change [ ] to [x]. -->

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 🔨 chore
- [ ] 📝 docs

### Relevant Issues

<!-- Use "resolves #xxx" to auto resolve on merge. Otherwise, please use "connect #xxx" -->

resolves #666 


### What is in this change?

Describe the changes in this PR that are impactful to the repo.

- Implement an admin view to see all logs coming from our AnythingLLM instance
- Implement export to csv
- Show logs for the following events:
  - API request
  - Workspace CRUD action
  - Filesystem Change
  - Chat Sent
  - Login events
  - Attempted Login events with IP

### Additional Information

Add any other context about the Pull Request here that was not captured above.

### Developer Validations

<!-- All of the applicable items should be checked. -->

- [ ] I ran `yarn lint` from the root of the repo & committed changes
- [ ] Relevant documentation has been updated
- [ ] I have tested my code functionality
- [ ] Docker build succeeds locally

timothycarambat reviewed (changes requested):
I think this is a good start but right now, this logging does not really help an admin have a full overview of a system and its activities.

Here is a more comprehensive list of tracking points to be added:
- System preference was modified (from => to value and setting)
- Invitation was created or deleted
- user was created, updated, suspended, or removed
- workspace deleted (via frontend or admin setting)
- chat logs exported
- API key created or deleted. I think created already exists
- Modification of LLM, Embedder, or Vector DB provider
- Embed chat created, updated, disabled, or deleted

Other items
- We can remove `extension_invoked` call logs
- The user should be able to "Reset event logs" from the UI, which deletes all existing event logs, but also at the end produces a log event that this was done. This endpoint does not exist currently.

shatfield4 commented:
- [ ] System preference was modified (from => to value and setting)
- [ ] Invitation was created or deleted
- [ ] user was created, updated, suspended, or removed
- [ ] workspace deleted (via frontend or admin setting)
- [ ] chat logs exported
- [ ] API key created or deleted. I think created already exists
- [ ] Modification of LLM, Embedder, or Vector DB provider
- [ ] Embed chat created, updated, disabled, or deleted
